doc/admin-guide/files/records.yaml.en.rst

```diff
@@ -4254,6 +4254,14 @@ OCSP Stapling Configuration
 
    Number of seconds before an OCSP response expires in the stapling cache.
 
+.. ts:cv:: CONFIG proxy.config.ssl.ocsp.request_mode INT 0
+
+   Set the request method to prefer when querying OCSP responders. The default
+   is zero, or POST, and a value of 1 will cause ATS to attempt a GET request.
+   Because the length of the encoded request must be less than 255 bytes per RFC
+   6960, Appendix A, ATS will fall back to the POST request method when the
+   encoded size exceeds this limit.
+
```

Comment:
As it is, it seems like it would be clearer to name this "prefer_get_method" or some such, and 1 enables it. Are you anticipating other possible configurations that might be set for this?

Reply:
I don't have any plans to add another mode at the moment. A possible mode would be "Always try GET regardless of the request size". The documentation uses "must" for the size limitation, but strictly speaking it's not a "MUST" on the RFC.

>   HTTP-based OCSP requests can use either the GET or the POST method to
   submit their requests.  To enable HTTP caching, small requests (that
   after encoding are less than 255 bytes) MAY be submitted using GET.
   If HTTP caching is not important or if the request is greater than
   255 bytes, the request SHOULD be submitted using POST.

I'm not sure if there are OCSP responders that accept big GET requests, but if some of them do, it can be an option.

@elsloo What do you think?

Reply:
I asked myself the same question when I worked on this initially, and had originally named the configuration parameter `request_method` and later renamed to `request_mode` to be more generic. I agree that the RFC is mostly clear on the behavior here but it isn't absolutely clear (SHOULD vs MUST), and as a result, we may in the future need another option to support a mixed mode of some kind bound by a size check to switch between the two request methods dynamically. I don't think that we should let that concern override the improvements here, and if someone does have GET requests that exceed the RFC-defined length, they can a) develop the support for variable request methods based on size which would require a new mode in configuration, or b) fall back to using POST for all requests.

I would personally prefer to keep this variable generic due to the RFC, which allows us to retain the variable if we need to extend it in the future. This is also somewhat of an advanced configuration topic, and most operators will likely read the documentation if they have any reason to move from POST to GET.

Reply:
Sounds good.